docs/_build/html/_sources/methods.rst.txt

```diff
@@ -0,0 +1,444 @@
+
+  % create a synapse using the cpplab object 'syn_cpplab'
+  x.connect('AB', 'LP', syn_cpplab)
+
+The following properties can be specified
+
+======================= ================
+Name                    ``PropertyName``
+Maximal conductance     ``gbar``
+Reversal potential      ``E``
+Activation variable     ``s``
+======================= ================
+=======
+Connects two compartments with a synapse. This defaults to an electrical synapse with axial conductance of ``NaN``.
+
+Connect two compartments ``AB`` and ``PD`` with an axial conductance of ``NaN`` ::
+
+  x.connect('AB', 'PD')
+
+Connect two compartments with an axial conductance of ``gAxial`` ::
+
+  x.connect('AB', 'PD', gAxial)
+
+Connect two compartments with a glutamatergic synapse with maximal conductance ``gGlut`` ::
+
+  x.connect('AB', 'PD', 'Glut', gGlut)
+>>>>>>> 2c44bd81aed5d11aac7fb1677a4b462d47822744
+
+.. _copy:
+
+copy
+^^^^
+
+Copies the entirety of a ``cpplab`` object into a new variable. ::
+
+  % create a new xolotl object
+  N = x.copy
+  % create a new HH object
+  N = x.HH.copy
+
+.. _get:
+
+get
+^^^
+::
+
+  values = x.get('findString')
+
+Returns a vector of doubles of the values stored in the specified fields. Automatically calls the ``find`` function; the argument is a search query for the ``find`` function. ::
+
+  % find all maximal conductances
+  gbars = x.get('*gbar')
+  % find all maximal conductances in HH compartment
+
+Produces a vector which contains the values of specified properties. ``get`` implicitly calls ``find`` and uses it to identify and order the elements of the vector. For this reason, ``get`` uses wildcard string comprehension. For example, ``'HH*gbar'`` means anything that begins with ``'HH'`` and ends with ``'gbar'``.
+Find all the maximal conductances of the ``HH`` compartment ::
+
+  gbars = x.get('HH*gbar')
+
+.. _find:
+
+find
+^^^^
+
+::
+
+  x.find('findString')
+
+Returns a cell array of character vectors for a search query. This function is a ``cpplab`` method, so what it does is specific to in which scope it is called.
+For example, for a xolotl object ``x`` with a compartment ``HH`` with three conductances ``NaV, Kd, Leak`` ::
+
+  x.find('*gbar')
+
+returns ::
+
+  {'HH.Kd.gbar'  }
+  {'HH.Leak.gbar'}
+  {'HH.NaV.gbar' }
+
+whereas, a call within the ``HH`` scope ::
+
+  x.HH.find('*gbar')
+
+yields ::
+
+  {'Kd.gbar'  }
+  {'Leak.gbar'}
+  {'NaV.gbar' }
+
+The ``*`` symbol is the wildcard operator. It finds any string that matches that pattern (e.g. ``*gbar`` finds any string in the ``xolotl`` tree that ends with ``'gbar'``).
+For example, ``x.find('HH*gbar')`` would only fetch paths to maximal conductances in the ``HH`` compartment.
+
+.. hint::
+
+  The ``set`` and ``get`` functions use the same syntax as the ``find`` function. They call it implicitly.
+
+.. _getGatingFunctions:
+
+getGatingFunctions
+^^^^^^^^^^^^^^^^^^
+
+::
+
+  [act ict tau_act tau_ict] = xolotl.getGatingFunctions('conductancePath')
```

Comment:
users are discouraged from using this. instead, they should look for the functions in the object tree directly, e.g.:

x.AB.NaV.cpp_child_functions

Reply:
actually never mind, this is fine. but also point out that this is already there in the object tree